src/community/gsr/src/main/java/org/geoserver/gsr/api/map/MapServiceController.java

```diff
@@ -72,18 +73,26 @@ public MapServiceRoot mapServiceGet(
             @PathVariable String workspaceName, @PathVariable String layerName) throws IOException {
         WorkspaceInfo workspace = geoServer.getCatalog().getWorkspaceByName(workspaceName);
         if (workspace == null) {
-            throw new NoSuchElementException(
-                    "Workspace name " + workspaceName + " does not correspond to any workspace.");
+            throw new APIException(
+                    "InvalidWorkspaceName",
+                    workspaceName + " does not correspond to any valid workspaces.",
```

Comment:
I don't think we need the "valid"

```suggestion
                    workspaceName + " does not correspond to any workspaces.",
```